logstash-core/lib/logstash/runner.rb

```diff
@@ -162,7 +171,11 @@ def execute
     end
 
     @agent = create_agent(:logger => @logger,
-                          :auto_reload => @auto_reload)
+                          :auto_reload => @auto_reload,
+                          :collect_metric => true,
+                          :debug => debug?,
```

Comment:
as mentioned already, will there ever be a situation where logger.debug? != debug?